nautobot/core/models/querysets.py

```diff
@@ -6,17 +6,23 @@
 from nautobot.core.utils.data import merge_dicts_without_collision
 
 
-def count_related(model, field):
+def count_related(model, field, filter_dict=None):
```

Comment:
As a nit, we might want to make filter_dict a keyword-only argument for clarity? I.e.:

```suggestion
def count_related(model, field, *, filter_dict=None):
```

and

```python
debug_log_count=count_related(JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_DEBUG}),
```

etc.